src/module_type_translator.ts

```diff
@@ -472,29 +472,35 @@ export class ModuleTypeTranslator {
 
       // Return type.
       if (!isConstructor) {
+        const returnTag: jsdoc.Tag = {
+          tagName: 'return',
+        };
         const retType = typeChecker.getReturnTypeOfSignature(sig);
-        const retTypeString: string = this.typeToClosure(fnDecl, retType);
-        let returnDoc: string|undefined;
-        for (const {tagName, text} of tags) {
-          if (tagName === 'return') {
-            returnDoc = text;
-            break;
+        // tslint:disable-next-line:no-any accessing TS internal field.
+        if ((retType as any).isThisType) {
```

Comment:
This is probably fine, but it also seems likely there's some way to get at the 'this' without private API, like getting flags from it.

Reply:
There isn't, I've looked quite a bit. The type then only looks like a parameterized type otherwise.